ng/src/web/components/dialog/linkconfirmationdialog.js

```diff
@@ -0,0 +1,176 @@
+ */
+
+import React from 'react';
+import glamorous from 'glamorous';
+
+import _ from 'gmp/locale.js';
+
+import {is_defined} from 'gmp/utils.js';
+
+import PropTypes from '../../utils/proptypes.js';
+
+import Dialog from '../dialog/dialog.js';
+import DialogContent from '../dialog/content.js';
+import DialogTitle from '../dialog/title.js';
+import ScrollableContent from '../dialog/scrollablecontent.js';
+
+import Button from '../dialog/button.js';
+
+import Layout from '../../components/layout/layout.js';
+
+const StyledLayout = glamorous(Layout)({
+  display: 'flex',
+  justifyContent: 'space-between',
+  borderWidth: '1px 0 0 0',
+  borderStyle: 'solid',
+  borderColor: '#ddd',
+  marginTop: '15px',
+  padding: '10px 15px 10px 15px',
+});
+
+class ConfirmationDialogContent extends React.Component {
+
+  constructor(...args) {
+    super(...args);
+
+    this.handleResume = this.handleResume.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  handleResume(url) {
+    const {onResumeClick} = this.props;
+
+    if (onResumeClick) {
+      const promise = onResumeClick(url);
+      if (is_defined(promise)) {
+        promise.then(
+          () => this.handleClose()
+        );
+      }
+      else {
+        this.handleClose();
+      }
+    }
+  }
+
+  handleClose() {
+    const {close} = this.props;
+    close();
+  }
+
+  render() {
+    let {
+      moveprops,
+      text,
+      title,
+    } = this.props;
+
+    // display slashes correctly, which have been escaped before
+    text = text.replace(/&#x2F;/g, '/');
+
+    return (
+      <DialogContent>
+        <DialogTitle
+          title={title}
+          onCloseClick={this.handleClose}
+          {...moveprops}
+        />
+        <ScrollableContent>
+          {text}
+        </ScrollableContent>
+        <StyledLayout>
+          <Button
+            onClick={this.handleClose}
+            title={_('Cancel')}
+          >
+            {_('Cancel')}
+          </Button>
+          <Button
+            onClick={this.handleResume}
+            title={_('Resume')}
+          >
+            {_('Resume')}
+          </Button>
+        </StyledLayout>
+      </DialogContent>
+    );
+  }
+}
+
+ConfirmationDialogContent.propTypes = {
+  close: PropTypes.func.isRequired,
+  moveprops: PropTypes.object,
+  text: PropTypes.string,
+  title: PropTypes.string.isRequired,
+  onResumeClick: PropTypes.func.isRequired,
+};
+
+const LinkConfirmationDialog = ({
+  children,
+  width,
+  text,
+  title,
+  to,
+  visible,
+  onClose,
+  onResumeClick,
+}) => {
+  return (
+    <Dialog
+      visible={visible}
+      width={width}
+      onClose={onClose}
+    >
+      {({
+        close,
+        getMoveProps,
+      }) => (
+        <ConfirmationDialogContent
+          close={close}
+          moveprops={getMoveProps()}
+          text={text}
+          title={title}
+          onResumeClick={() => onResumeClick(to)}
```

Comment:
Something is wrong with the onResumeClick handler. I guess you did mix something. Here it would be better to pass the to prop also to ConfirmationDialogContent and not to create a arrow handler function.